solidity/dashboard/src/sagas/staking.js

```diff
@@ -178,3 +191,72 @@ function* fetchTopUps() {
     yield* logError("staking/fetch_top_ups_failure", error)
   }
 }
+
+export function* watchTopUpReadyToBeCommitted() {
+  // Waiting for top-ups data.
+  yield take("staking/fetch_top_ups_success")
+  yield call(notifyTopUpReadyToBeCommitted)
+
+  while (true) {
+    // Every 5 minutes.
+    yield delay(moment.duration(5, "minutes").asMilliseconds())
+    yield call(notifyTopUpReadyToBeCommitted)
+  }
+}
+
+function* notifyTopUpReadyToBeCommitted() {
+  const topUps = yield select((state) => state.staking.topUps)
+  const initializationPeriod = yield select(
+    (state) => state.staking.initializationPeriod
+  )
+
+  const topUpsReadyToCommit = topUps.filter(({ createdAt }) =>
+    moment
+      .unix(createdAt)
+      .add(initializationPeriod, "seconds")
+      .isBefore(moment())
+  )
```

Comment:
I think there is a problem with the top-ups (unless this is how it should work, but probably not). When you stake some tokens for a new operator, that is not present in top-ups table yet, the new row in top-ups will be created  but the commit top-up button is disabled. The notification that it's ready to be committed does not appear also. After refresh and connecting with the wallet again, everything works as expected.

Reply:
So the first problem is with notifications - they do not appear for newly created top-ups and i managed to debug the code to find the issue. The top-ups that already exist in the table have type of the `createdAt` field as the number, while newly created top-ups have it as `Moment` object. It is changed to number after we relog and top ups update from fetching. This is why, for newly created top-ups, the notification does not appear - the `topUpsReadyToCommit` variable above does not filter the topUps that have `createdAt` fiels as `Moment` object properly. We can resolve this by changing this line of code:

https://github.com/keep-network/keep-core/blob/0c5748cd54e8f479a8ffbeedb2a17fe06264591b/solidity/dashboard/src/reducers/staking.js#L153

to `moment().unix()`. There is also similar issue with the code that is a few lines below:

https://github.com/keep-network/keep-core/blob/0c5748cd54e8f479a8ffbeedb2a17fe06264591b/solidity/dashboard/src/reducers/staking.js#L163

P.S.

There is also similar problem with `availableTopUpAmount` here:
https://github.com/keep-network/keep-core/blob/0c5748cd54e8f479a8ffbeedb2a17fe06264591b/solidity/dashboard/src/reducers/staking.js#L148-L157

For newly created top-ups the value of it is in **string** while for the fetched top-ups it is **BN**. I haven't noticed any issues with it, but it probably needs investigation.